### PR TITLE
[FLINK-34931] Update Sink API to V2

### PR DESCRIPTION
**Pull request by ferenc-csaky**
* Change `KuduWriter` to implement the `SinkWriter` interface.
* Introduce V2 `KuduSink` and a builder class.
* Adapt `KuduOutputFormat`.
* Remove deprecated Table API classes and tests (TableSource, TableSink), as TableSink will not be compatible with the new Sink, and the connector already contains the dynamic table stack impl as well.
* Adapt existing tests.

### CODE REVIEW

File: flink-connector-kudu/src/main/java/org/apache/flink/connector/kudu/connector/writer/KuduWriter.java [hunk 1]
```diff
@@ -35,11 +36,12 @@
 
 import java.io.IOException;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 /** Writer to write data to a Kudu table. */
 @Internal
-public class KuduWriter<T> implements AutoCloseable {
+public class KuduWriter<T> implements SinkWriter<T>, AutoCloseable {
```
Comment (mateczagany) [inline flink-connector-kudu/src/main/java/org/apache/flink/connector/kudu/connector/writer/KuduWriter.java:44]: `AutoClosable` is redundant and can be removed

File: flink-connector-kudu/src/main/java/org/apache/flink/connector/kudu/sink/KuduSink.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        this.operationMapper = operationMapper;
+        this.failureHandler = failureHandler;
+    }
+
+    public static <IN> KuduSinkBuilder<IN> builder() {
+        return new KuduSinkBuilder<>();
+    }
+
+    @Override
+    public SinkWriter<IN> createWriter(InitContext initContext) throws IOException {
```
Comment (mateczagany) [inline flink-connector-kudu/src/main/java/org/apache/flink/connector/kudu/sink/KuduSink.java:66]: According to the JavaDocs, `SinkWriter<InputT> createWriter(WriterInitContext context)` should also be implemented here.

**Review by mateczagany — COMMENTED** [review]
Just 2 small comments, otherwise all looks good.

File: flink-connector-kudu/src/main/java/org/apache/flink/connector/kudu/connector/writer/KuduWriter.java (hunk 1 above)
Comment (ferenc-csaky) [reply flink-connector-kudu/src/main/java/org/apache/flink/connector/kudu/connector/writer/KuduWriter.java]: Good catch!

File: flink-connector-kudu/src/main/java/org/apache/flink/connector/kudu/sink/KuduSink.java (hunk 2 above)
Comment (ferenc-csaky) [reply flink-connector-kudu/src/main/java/org/apache/flink/connector/kudu/sink/KuduSink.java:67]: When the deprecated method will be removed the impl has to be changed anyways, but fair point, did not read the javadoc.

**Review by mateczagany — APPROVED** [review]

**Pull request merged**